### DISCUSSION TITLE
Оптимизация аватара

### DISCUSSION THREAD

**Discussion by omqmo** [taiga-family/taiga-ui#5536] [Ideas]
При знакомстве с tui avatar меня смутило то, что картинки нельзя оптимизировать. Но как только я заметил новую секцию документацию "Эксперементы", в которой также присутствует аватарка и все также без возможности оптимизации, меня уже потянуло задать этот вопрос - у вас аватары реально нельзя оптимизировать или это я чего-то не понимаю? Под оптимизацией я понимаю ngSrc или что-то свое.

**Comment by waterplea** [comment]
В Angular 12 ещё нет ngSrc. В аватарке присутствует лейзи лоадинг с фолбэком на IntersectionObserver. Думаешь имеет смысл оптимизировать картинку 48х48 пикселей? Мне казалось это всё больше для крупных изображений актуально. Наверное можно добавить вот такой режим, тогда, для внешней оптимизации:
```html
<tui-avatar>
  <img ngSrc="..." />
</tui-avatar>
```

**Reply by splincode** [reply]
А захотят ли пользователи упороться и использовать picture?

```html
<tui-avatar>
    <picture>
      <source srcset="/media/cc0-images/surfer-240-200.jpg" media="(orientation: portrait)" />
      <img src="/media/cc0-images/painted-hand-298-332.jpg" alt="" />
    </picture>
</tui-avatar>
```

**Reply by omqmo** [reply]
> Думаешь имеет смысл оптимизировать картинку 48х48 пикселей? 

Если вы рассуждаете в контексте 48х48, то зачем тогда реализовали размер xxl, который 128х128 и для pixel ratio х2 уже 256х256. К тому же, есть админка с листом товаров, которых одновременно 10-15 во вьюпорте, они по 64х64. Получает, даже при десяти на экране, это холст 640х640, а с оптимизацией на px ratio x2, это 1280х1280. И это только на одну прокрутку.

> Наверное можно добавить вот такой режим, тогда, для внешней оптимизации:

Если бы вы такое реализовали, то было бы вообще великолепено!

**Reply by waterplea** [reply]
Поддержка внешней оптимизации картинок добавлена:
https://taiga-ui.dev/experimental/avatar#content
